x-pack/plugins/security/common/licensing/index.mock.ts

```diff
@@ -7,15 +7,27 @@
 
 import { of } from 'rxjs';
 
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { LICENSE_TYPE } from '../../../licensing/server';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import type { LicenseType } from '../../../licensing/server';
 import type { SecurityLicenseFeatures } from './license_features';
 import type { SecurityLicense } from './license_service';
 
 export const licenseMock = {
-  create: (features?: Partial<SecurityLicenseFeatures>): jest.Mocked<SecurityLicense> => ({
+  create: (
+    features: Partial<SecurityLicenseFeatures> = {},
+    licenseType: LicenseType = 'basic' // default to basic if this is not specified
+  ): jest.Mocked<SecurityLicense> => ({
     isLicenseAvailable: jest.fn().mockReturnValue(true),
     isEnabled: jest.fn().mockReturnValue(true),
-    getType: jest.fn().mockReturnValue('basic'),
-    getFeatures: jest.fn(),
+    getFeatures: jest.fn().mockReturnValue(features),
+    hasAtLeast: jest
+      .fn()
+      .mockImplementation(
+        (licenseTypeToCheck: LicenseType) =>
+          LICENSE_TYPE[licenseTypeToCheck] <= LICENSE_TYPE[licenseType]
+      ),
     features$: features ? of(features as SecurityLicenseFeatures) : of(),
   }),
 };
```

Comment:
I don't love importing `LICENSE_TYPE` directly from the licensing server for this securityLicenseService mock, but it seemed to be the least-bad option here.